### ISSUE TITLE
ECE integration

### ISSUE THREAD

**Issue by Bharat-Goyal** [open-mmlab/mmsegmentation#2629]
Would anyone happen to know the best way to integrate expected calibration error into the repository? Whenever we get the data for the metrics it doesnt have a probabilites distribution for the pixels but rather the classwise total (in case its pre eval) or the pixel label (in case its a call to eval_metrics). The only potential way to access the probabilities seems to be to get access to the immediate forward pass and results but then I don’t understand how to do that. I’d appreciate any pointers (I’ll get around to the implementation myself so that’s not a concern) from the more experienced members!

**Comment by MeowZheng** [comment]
Would you like to attach the paper about ECE? We might support it if there is large requirements about this feature.

**Comment by Bharat-Goyal** [comment]
https://www.dbmi.pitt.edu/wp-content/uploads/2022/10/Obtaining-well-calibrated-probabilities-using-Bayesian-binning.pdf 

Expected Calibration Error ^ 
It basically checks the model's predicted probability distribution with the ground truth probability distribution. Would you have any suggestions on a  way I could access those probabilities through the model?

**Comment by MeowZheng** [comment]
I'm not familiar with 'Expected Calibration Error', but I knew MC dropout is a very easy way to get model probability 
FYI: https://arxiv.org/abs/1506.02142

**Comment by Bharat-Goyal** [comment]
Won't the probabilities be calculated during the forward pass of the model though?

**Comment by Bharat-Goyal** [comment]
Oh so when I said probabilities I actually just meant the softmax values calculated by the model. The paper is definitely a good read, but prior to being able to use any of that I'll need to access the logits and I was wondering if there's a non hacky way to do that (right now I'm thinking of just modifying some of the base class functions to do that).

**Comment by xiexinch** [comment]
I think you might try to add an `after_train_iter ` hook, then it's no need to modify the base class.

**Comment by xiexinch** [comment]
Closing the issue, as there is no activity for a while.
We hope your issue has been resolved. 
If not, please feel free to open a new one.
